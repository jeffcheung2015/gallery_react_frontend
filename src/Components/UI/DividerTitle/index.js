import React, { Component } from 'react';
import "./index.scss"
import { Divider, Header, Icon } from 'semantic-ui-react'
import { pure } from 'recompose';
import PropTypes from 'prop-types';

const DividerTitle = (props) => {
  const {
    headerClassName, title, iconName
  } = props
  return (
    <Divider horizontal>
      <Header as='h4' className={headerClassName}>
        <Icon name={iconName} />
        {title}
      </Header>
    </Divider>
  )
}

export default pure(DividerTitle)
DividerTitle.propTypes = {
  title: PropTypes.string.isRequired,
  headerClassName: PropTypes.string,
  iconName: PropTypes.string.isRequired
}
