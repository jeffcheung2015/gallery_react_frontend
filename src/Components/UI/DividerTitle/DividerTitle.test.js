import React, { Component } from 'react'
import _times from 'lodash/times'
import renderer from 'react-test-renderer';
import DividerTitle from 'Components/UI/DividerTitle';
import toJson from 'enzyme-to-json'


let wrapper

let dataProps = {headerClassName: "title-class", iconName: "images outline", title: "i am title"}

describe("<DividerTitle />", () => {
  it("input -> correct output test", () => {
    const tree = shallow(
      <DividerTitle
        {...dataProps}
      />
    ).dive()
    expect(toJson(tree)).toMatchSnapshot();
  })
})
