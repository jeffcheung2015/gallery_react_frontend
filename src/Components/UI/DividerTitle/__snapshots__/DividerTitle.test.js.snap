// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DividerTitle /> input -> correct output test 1`] = `
<Divider
  horizontal={true}
>
  <Header
    as="h4"
    className="title-class"
  >
    <Icon
      as="i"
      name="images outline"
    />
    i am title
  </Header>
</Divider>
`;
